src/main/java/com/amazon/opendistroforelasticsearch/knn/index/KNNMethodContext.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package com.amazon.opendistroforelasticsearch.knn.index;
+
+import com.amazon.opendistroforelasticsearch.knn.index.util.KNNEngine;
+import org.opensearch.common.xcontent.ToXContentFragment;
+import org.opensearch.common.xcontent.XContentBuilder;
+import org.opensearch.index.mapper.MapperParsingException;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.Map;
+
+import static com.amazon.opendistroforelasticsearch.knn.common.KNNConstants.KNN_ENGINE;
+import static com.amazon.opendistroforelasticsearch.knn.common.KNNConstants.METHOD_HNSW;
+import static com.amazon.opendistroforelasticsearch.knn.common.KNNConstants.METHOD_PARAMETER_SPACE_TYPE;
+import static com.amazon.opendistroforelasticsearch.knn.common.KNNConstants.NAME;
+import static com.amazon.opendistroforelasticsearch.knn.common.KNNConstants.PARAMETERS;
+
+/**
+ * KNNMethodContext will contain the information necessary to produce a library index from an Opensearch mapping.
+ * It will encompass all parameters necessary to build the index.
+ */
+public class KNNMethodContext implements ToXContentFragment {
+
+    private static KNNMethodContext defaultInstance = null;
+
+    public static synchronized KNNMethodContext getDefault() {
```

Comment:
Do we need ```synchronized``` here? Will KNNMethodContext object be shared?

Reply:
My thinking is that this static method may be called from different threads to get the default.

Reply:
makes sense.